classes/services/SubmissionService.inc.php

```diff
@@ -100,7 +100,8 @@ public function addToCatalog($submission) {
 		$submission->setStatus(STATUS_PUBLISHED);
 		\Application::getSubmissionDao()->updateObject($submission);
 
-		$publishedMonograph->setDatePublished(\Core::getCurrentDate());
+		$datePublished = $submission->getDatePublished() ? $submission->getDatePublished() : \Core::getCurrentDate();
+		$publishedMonograph->setDatePublished($datePublished);
```

Comment:
Isn't submission->get/setDatePublished() = $publishedMonograph->get/setDatePublished() ? If so, then maybe just to set the date to the current date if there is not date?

Reply:
Maybe I'm confusing myself, but I think they're different objects here. `$publishedMonograph` is either pulled from the database based on the submission ID or a new `$publishedMonograph` is created. In either case, the `$submission` could have a new value, and if so this should be used to update the `$publishedMonograph`. Is that right?

Reply:
Ah ok, sorry, I thought wrong :-)

Reply:
please merge if ready :-)